### PR TITLE
Feature/72 saving draft form and clear response

### PR DESCRIPTION
**Pull request by wayangalihpratama**
### TODO / DONE

---

- #72 
- [x] Handle onSave callback on FormContainer
- [x] Add Kebab Menu to show "Save and Exit" / "Exit without Saving" button
  - [x] Dialog component
  - [x] Dropdown component 
- [x] Handle "Save and Exit" event (refresh form)
- [x] Handle "Exit without Saving" event (refresh form)
- [x] Add Back Button confirmation

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204868663603993
  - https://app.asana.com/0/0/1205117828000077

### CODE REVIEW

File: app/src/form/FormContainer.js [hunk 1]
```diff
@@ -11,7 +11,29 @@ import { FormState, UIState } from '../store';
 // TODO:: Allow other not supported yet
 // TODO:: Repeat group not supported yet
 
-const FormContainer = ({ forms, initialValues = {}, onSubmit }) => {
+const checkValuesBeforeCallback = (values) =>
+  Object.keys(values)
+    .map((key) => {
+      let value = values[key];
+      // check empty
+      if (
```
Comment (dedenbangkit) [inline app/src/form/FormContainer.js]: @wayangalihpratama I think we can simplify this.

```bash
λ node
Welcome to Node.js v18.7.0.
Type ".help" for more information.
> const values = [false, undefined, null, 0, NaN,'']
undefined
> values.map(v => !v)
[ true, true, true, true, true, true ]
```

But from what I see here, you want to keep 0 as valid value, so:

```jsx
if (value === 0) {
   return {[key]: value}
}
if (!value) {
   return false
}
```

File: app/src/form/__test__/FormContainer.test.js [hunk 2]
```diff
@@ -219,7 +219,53 @@ jest.mock('../../assets/administrations.db', () => {
...
+      expect(handleOnSave).toHaveBeenCalledTimes(1);
+      expect(handleOnSave).toHaveBeenCalledWith(null, expect.any(Function));
+    });
+  });
+
+  test('should handle onSave event and return refreshForm', async () => {
+    const handleOnSave = jest.fn();
+
+    const modifiedInitialValues = {
+      1: 'John',
```
Comment (dedenbangkit) [inline app/src/form/__test__/FormContainer.test.js:238]: Add more tests to make this empty string

**Review by dedenbangkit — CHANGES_REQUESTED** [review]

File: app/src/form/FormContainer.js (hunk 1 above)
Comment (wayangalihpratama) [reply app/src/form/FormContainer.js]: Hi mas @dedenbangkit , I pushed an update for this

File: app/src/form/__test__/FormContainer.test.js (hunk 2 above)
Comment (wayangalihpratama) [reply app/src/form/__test__/FormContainer.test.js:238]: Hi mas @dedenbangkit , I pushed an update for this

**Review by dedenbangkit — APPROVED** [review]
